docs/compatibility.md

```diff
@@ -0,0 +1,185 @@
+- 1/29/23: Remove warning in libStablehlo@v3
+  + Backward compatibility: Nothing changes from 1/7/2023.
+  + Forward compatibility: Window has closed, no need to emit warning anymore.
+
+### Deprecation Change
+_Characteristics of a deprecation change:_
+- Remove a feature with backwards compatibility
+- Prevent creation of new bytecode using feature
+- Clean up after backward compatibility window
+
+_Example:_ Change enum `Enum` which has values `<A>` and `<B>` to remove value `<B>`.
+
+_Compatibility Protocol:_
+
+![Deprecation Change Protocol Diagram](images/deprecation_change.png)
+
+_Description:_
+- 1/1/23: Create bytecode_v1 from libStablehlo@v1
+
+- 1/7/23: Add compatibility check to prevent serialization of `<B>` in libStablehlo@v2
+  + Backward compatibility: libStablehlo@v2 can deserialize `<A>` and `<B>` from bytecode_v1.
+  + Forward compatibility: libStablehlo@v1 can handle `<A>` from bytecode_v2. libStablehlo@v2 prevents the serialization of `<B>`.
+- 6/8/23: Remove `<B>` deserialization and enum value in libStablehlo@v3.
+  + Ensure that numeric value of other enums does not change due to deletion.
+  + Backward compatibility: Window has closed, no need to maintain `<B>`.
+  + Forward compatibility: No changes required, libStablehlo@v2 can handle all libStablehlo@v3 programs.
+
+### Versioned Op Change
+_Characteristics of a change that requires versioning:_
+- A change is introduced that cannot be upgraded / downgraded
+- Create new version of legacy op
+- A combination of Addition/Deprecation Protocols from there
+
+_Example:_ Changing the signature and semantics to an op, adding an operand.
+
+_Compatibility Protocol:_
+
+![Versioned Op Change Protocol Diagram](images/versioned_op_change.png)
+
+_Description:_
+- 1/1/23: Create bytecode_v1 from libStablehlo@v1
+- 1/7/23: Add compatibility check to prevent serialization of `MyOp` in libStablehlo@v2
+  + Add compatibility check to warn if `MyOp_v2` is used and cannot be downgraded.
+  + No change in serialization so still forward compatible
+- 1/29/23: Remove forward compatibility warning 
+- 6/8/23: Remove `MyOp` in libStablehlo@v4
+  + Backward compatibility window over.
+  + Optionally rename `MyOp_v2` to `MyOp` via Migration Protocol
+
+### Changes in Dependent Dialects
+StableHLO will use forks of supported operations in dependent dialects for serialization / deserialization only. These forked operations allow StableHLO a finer control of the compatibility changes of operations in other dialects using the protocols above.
+
+_Proposal:_
+- Have stable shims of supported operations in other dialects.
+- Downgrade IR to shim before serializing `arith.addi → stablehlo.arith.addi`
+- Upgrade IR to other dialect on deserialization.
+
+With the above rules, we are able to provide compatibility guarantees for Migration and Additions. 
+
+_Example:_ Move `builtin.func → func.func`.
+
+![Dependent Dialect Change Protocol Diagram](images/dependent_dialect_change.png)
```

Comment:
The diagram still says "stable.builtin.func". Rename to "stablehlo.builtin.func"?